### ISSUE TITLE
Home - Sticky Nav Bar

### ISSUE THREAD

**Issue by aryamankha** [aryamankha/airbnb-clone#2]
<img width="897" alt="Screen Shot 2022-10-11 at 6 21 04 PM" src="https://user-images.githubusercontent.com/62365335/195209069-833b7329-5b91-420b-8a7a-5db920886d51.png">

Implement sticky nav bar using Vivid! In the browser, command-click on any component to edit its code and hit Command-K to quickly preview styles. Command-Shift-K will let you add new empty components!

**Comment by Alphx-rgb** [comment]
I'll work on this.

**Comment by aryamankha** [comment]
Great - assigning to you

**Comment by Alphx-rgb** [comment]
I have almost created the component  but stuck on some errors . Should I update the code for everybody to view?

**Comment by aryamankha** [comment]
What are the errors you found?

**Comment by Rishabh2324** [comment]
Is this issue fixed...  ?

**Comment by Alphx-rgb** [comment]
yeah, this is the error I am facing too... I am checking several resources but have not been able to figure out a solution.  I guess we can create a new issue for this. In the meanwhile, I'll try myself also for correcting the same. And after this issue gets fixed, We can focus on building more components.

**Comment by Rishabh2324** [comment]
Take reference from this 

1. First update DataContext

`export const initialState: IInitialState = {
  location: '',
  checkIn: null,
  checkOut: null,
  guests: { adults: 0, children: 0, infants: 0 },
};

export const DataContext = createContext<{
  state: IInitialState;
  dispatch: React.Dispatch<any>;
}>({
  state: initialState,
  dispatch: () => null,
});



interface IIProps {
  children: React.ReactNode;
}
export const ContextProvider = ({ children }: any) => {
  const [state, dispatch] = useReducer(dataReducer, initialState);
  return (
    <>
      <DataContext.Provider value={{ state, dispatch }}>
        {children}
      </DataContext.Provider>
    </>
  );
};
`

2. While using your context in Date range Desctruture like this
`
   const { state, dispatch } = useDataContext();
  const { location, checkIn, checkOut, guests } = state;

`

3. Also add ContextProvider to parent component so that dispatch can work
4. Avoid using any

**Comment by Alphx-rgb** [comment]
I am facing a few errors which I am unable to resolve. I'll work on it, if I get it, I'll make a PR.

**Comment by aryamankha** [comment]
Assigning it to you

**Comment by Hamjaster** [comment]
I've a question, what does Vivid in "Sticky Nav by using Vivid" means??

**Comment by Hamjaster** [comment]
I ran the code in my local environment, Theres just a simple navbar in it? what to do, ? I am quite confused
